### PR TITLE
Limit use of Axom CMake targets

### PR DESCRIPTION
**Pull request by white238**
This limits which Axom CMake targets are propagated to downstream users. This should only make `axom::primal` and below required to be propagated, while `axom::quest` is only used internally.

~~I also did a very quick pass-through to quiet all compiler warnings. I won't be offended if you don't like how I fixed them.~~

### CODE REVIEW

File: src/examples/CMakeLists.txt
Comment (ebchin) [inline src/examples/CMakeLists.txt:14]: Is this one still needed?

File: src/tribol/CMakeLists.txt
Comment (ebchin) [inline src/tribol/CMakeLists.txt:122]: 👍

File: src/tribol/physics/CommonPlane.cpp
Comment (ebchin) [inline src/tribol/physics/CommonPlane.cpp]: I think this should stay commented out for now @srwopschall?  Or potentially removed/moved to common plane tests?

File: src/tribol/utils/TestUtils.cpp
Comment (ebchin) [inline src/tribol/utils/TestUtils.cpp]: @srwopschall can you verify these changes are OK?  I think `SLIC_DEBUG_IF` seems like the right level here, but #39 will have to be addressed first.

**Review by ebchin — APPROVED** [review]
Thanks @white238!  The CMake changes look fine to me, but @srwopschall should verify the changes to the slic macros before merging.

File: src/examples/CMakeLists.txt
Comment (kennyweiss) [reply src/examples/CMakeLists.txt:14]: Yes. Unfortunately, the `cuda` dependency needs to be explicitly listed.

File: src/tests/tribol_mortar_data_weights.cpp
Comment (kennyweiss) [inline src/tests/tribol_mortar_data_weights.cpp]: It looks like this variable was used for debugging. @srwopschall -- would you prefer to keep it and the output below as comments?

File: src/tests/tribol_mortar_force.cpp
Comment (kennyweiss) [inline src/tests/tribol_mortar_force.cpp]: Perhaps this should be initialized to `UNDEFINED_ELEMENT`?

https://github.com/LLNL/Tribol/blob/892d27d21ebce6931d38354f76444c490e8f8042/src/tribol/common/Parameters.hpp#L49-L58

(here and elsewhere)

File: src/tribol/geom/GeomUtilities.cpp
Comment (kennyweiss) [inline src/tribol/geom/GeomUtilities.cpp]: (One line up) `diffTol` should be a constexpr:
```diff
- real diffTol = 1.0E-3;
+ constexpr real diffTol = 1.0E-3;
```

and perhaps the other vars should be `const`?

File: src/tribol/physics/CommonPlane.cpp
Comment (kennyweiss) [reply src/tribol/physics/CommonPlane.cpp]: I agree that these lines should stay commented out

File: src/tribol/utils/TestUtils.cpp
Comment (kennyweiss) [reply src/tribol/utils/TestUtils.cpp]: @white238 -- we were observing way too much output in a user application, so @srwopschall recently converted a lot of `INFO` logs to `DEBUG`. He's planning a quick follow-up PR to increase the verbosity where appropriate.

Tribol likely needs a user-controlled log verbosity level for INFO and possibly DEBUG log messages. Perhaps going from 0 (no logs) to 5 (all the logs).

**Review by kennyweiss — COMMENTED** [review]
Thanks for the fixes @white238!

I think @srwopschall will want to update the changes to the log levels to ensure they're not too verbose. He's already planning a follow-up PR to #29 focus on the log statements.

File: src/tests/tribol_mortar_data_weights.cpp
Comment (srwopschall) [reply src/tests/tribol_mortar_data_weights.cpp]: Yes, let's keep this variable and the output below that uses it as comments. @white238 can we restore this and comment out?

File: src/tribol/geom/GeomUtilities.cpp
Comment (srwopschall) [inline src/tribol/geom/GeomUtilities.cpp]: When/how is this defined now? Does this properly address issue https://github.com/LLNL/Tribol/issues/39?

File: src/tribol/geom/GeomUtilities.cpp
Comment (srwopschall) [reply src/tribol/geom/GeomUtilities.cpp]: This can be addressed when https://github.com/LLNL/Tribol/issues/36 is addressed.

File: src/tribol/physics/CommonPlane.cpp
Comment (srwopschall) [reply src/tribol/physics/CommonPlane.cpp]: Agree, please leave as commented out. How lines like this are handled with be addressed in future PR.

File: src/tribol/utils/TestUtils.cpp
Comment (srwopschall) [reply src/tribol/utils/TestUtils.cpp]: [src/tribol/geom/GeomUtilities.cpp](https://github.com/LLNL/Tribol/pull/40/files#diff-881a58c856789d0769a178217fdaa4b9fdd4e1f19e4293fdd5e68f50b07f7b6f) uses AXOM_DEBUG. Can we do that here? Is AXOM_DEBUG properly defined?

**Comment by white238** [comment]
I am going to revert the warnings commit and leave this to you guys to decide how you want the warnings fixed.

**Review by srwopschall — APPROVED** [review]

**Pull request merged**
